src/lib/translators/mongo.ts

```diff
@@ -105,23 +105,42 @@ Object.assign(Mongo36Translator.prototype, mapper);
  *
  * @returns the list of simplified mongo steps
  */
-function simplifyMongoPipeline(mongoSteps: Array<MongoStep>): Array<MongoStep> {
+export function _simplifyMongoPipeline(mongoSteps: Array<MongoStep>): Array<MongoStep> {
+  let merge: boolean = true;
   const outputSteps: Array<MongoStep> = [];
   let lastStep: MongoStep = mongoSteps[0];
   outputSteps.push(lastStep);
 
   for (const step of mongoSteps.slice(1)) {
-    if (step.$project !== undefined && lastStep.$project !== undefined) {
-      // merge $project steps together
-      lastStep.$project = { ...lastStep.$project, ...step.$project };
-      continue;
-    } else if (step.$match !== undefined && lastStep.$match !== undefined) {
-      // merge $match steps together
-      lastStep.$match = { ...lastStep.$match, ...step.$match };
-      continue;
+    const stepOperator: string = Object.keys(step)[0];
+    const isMergeable: boolean =
+      stepOperator === '$project' || stepOperator === '$addFields' || stepOperator === '$match';
+    if (isMergeable && lastStep[stepOperator] !== undefined) {
+      for (const key in step[stepOperator]) {
+        // We do not want to merge two $project with common keys
+        if (lastStep[stepOperator].hasOwnProperty(key)) {
+          merge = false;
```

Comment:
Shoudn't we `break` once `merge` is set to `false`?

Reply:
@adimascio I tried to use `break` but it seems a little tricky to use as it breaks out from parent loops to. So basically as soon as I met a `merge: false`case, my Mongo pipeline was over as I went out of the root loop (`for (const step of mongoSteps.slice(1))`). 

But open to discussion if I (probably) miss something !